core/modules/layout/layout.admin.inc

```diff
@@ -885,6 +892,33 @@ function layout_settings_form_validate(&$form, &$form_state) {
       form_error($form['conditions']['active'][$key], t('The condition \'!condition\' does not have the required contexts at this path. Remove the condition to save the layout.', array('!condition' => $condition->summary())));
     }
   }
+
+  global $_layout_additional_paths;
+  $_layout_additional_paths = array();
+  if (isset($form_state['values']['additional_paths'])) {
+    $primary_path = $form_state['values']['path'];
+    $num_placeholders = count(_layout_placeholder_positions($primary_path));
+    $paths = explode("\n", $form_state['values']['additional_paths']);
+    foreach($paths as $path){
+      $path = trim($path);
+      if(!empty($path)) {
+        if ($num_placeholders > count(_layout_placeholder_positions($path))) {
+          form_set_error('additional_paths', t('Additional path %path should have the same, or greater, number of placeholders as primary path %primary_path.',
+            array(
+              '%path' => $path,
+              '%primary_path' => $primary_path,
+            )
+          ));
+        }
+        elseif(_layout_validate_path($form, 'additional_paths', $path)) {
```

Comment:
Maybe Saturday is a bad day for code review, but I'm having some trouble to understand the logic here.

If the placeholder count matches, and the placeholder is correct (by syntax), ignore it (continue).
If there's no placeholder, or the placeholder is invalid, add it... :thinking: 
Am I missing something?

Another suggestion (just for readability) would be to name the var inside the loop `$additional_path` (instead of just $path) - so it's clear, what we're dealing with here. :wink:

Reply:
Maybe also rename the function `_layout_validate_path()` so it's clearer that it's setting the form error inside it?